.eslintrc.js

```diff
@@ -0,0 +1,37 @@
+module.exports = {
+  root: true,
+
+  extends: ['@metamask/eslint-config'],
+
+  parserOptions: {
+    sourceType: 'module',
+  },
+
+  overrides: [
+    {
+      files: ['*.ts'],
+      extends: ['@metamask/eslint-config-typescript'],
+    },
+    {
+      files: ['*.js', '*.ts'],
+      parserOptions: {
+        sourceType: 'script',
+      },
+      extends: ['@metamask/eslint-config-nodejs'],
+    },
```

Comment:
I meant to suggest something more like this:

```suggestion
  extends: ['@metamask/eslint-config', '@metamask/eslint-config-nodejs'],

  overrides: [
    {
      files: ['*.ts'],
      extends: ['@metamask/eslint-config-typescript'],
    },
    {
      files: ['*.js'],
      parserOptions: {
        sourceType: 'script',
      },
    },
```

Then we won't have to override the `sourceType` in so many different places (it's set in some of these shared configs as well).